openedx/core/djangoapps/content/search/plain_text_math.py

```diff
@@ -0,0 +1,105 @@
+"""
+Helper class to convert mathjax equations to plain text.
+"""
+
+import re
+
+import unicodeit
+
+
+class PlainTextMath:
+    """
+    Converts mathjax equations to plain text using unicodeit and some preprocessing.
+    """
+    equation_pattern = re.compile(
+        r'\[mathjaxinline\](.*?)\[\/mathjaxinline\]|\[mathjax\](.*?)\[\/mathjax\]|\\\((.*?)\\\)|\\\[(.*?)\\\]'
+    )
+    eqn_replacements = (
+        # just remove prefix `\`
```

Comment:
Bug: `\(x^2 + \sqrt(y)\)` (and `[mathjaxinline]x^2 + \sqrt(y)[/mathjaxinline]`) are rendered by mathjax in the block Preview like this: 
![image](https://github.com/user-attachments/assets/4ceaff94-4bf7-4f39-805a-9add1dc687ad)

But that `\sqrt` isn't being converted to `√` on the Card preview:

![image](https://github.com/user-attachments/assets/8e50cdf9-0595-4768-a855-982a871a9178)

Could we add another eqn/regex + test cases for this and make sure this syntax is being replaced too?
```suggestion
        # some equations aren't recognized by unicodeit
        ("\\sqrt", "√"),
        # just remove prefix `\`
```

Reply:
@pomegranited I am not sure if I understand the bug, `\sqrt` seems to be converted?

![image](https://github.com/user-attachments/assets/7891fe4f-2b6b-425f-8e96-45b2fb69630a)

Reply:
I also see it converted.

Reply:
Oh wow sorry.. the example code I was using didn't have the `\` before the `sqrt`, and so it wasn't converting. Sorry for the noise!